.github/workflows/on_push.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "master", "main" ]
+  pull_request:
+    branches: [ "master", "main" ]
+  workflow_dispatch:
+    branches: [ "master", "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
```

Comment:
It looks like GitHub has dropped support for running Python 2.7 in workflows:
https://github.com/actions/setup-python/issues/672
Someone in the thread proposed a workaround that I could try implementing:
https://github.com/actions/setup-python/issues/672#issuecomment-1589120020

Reply:
@ElijahCapps I saw your comment on Slack that GitHub workflows don't  support Python 2.7 builds. Let's discuss that here (not in the general channel on the ARFC slack).

What is the reasoning for moving from Travis to GitHub Actions anyway? (@munkm I'm sure there's a reason, I just haven't been in the loop for the last few years... Maybe you can explain why?)

Reply:
I believe Munk told me to move it from Travis to Actions because Actions is newer.

Reply:
It's not because actions is newer, it's because Travis removed their free runs and you now need to pay for CI jobs with them. Most projects had to migrate from Travis in 2020/2021 because of this issue.

Reply:
Oh, that makes a lot more sense.

Reply:
@katyhuff I'm looking at using images from DockerHub to run Python versions 2.7 and 3.10 in their own containers, but I'm unsure of which versions to pick. There are a lot available and it lists many as having vulnerabilities I'm not familiar with. What would you suggest?